src/bst.py

```diff
@@ -0,0 +1,111 @@
+"""This module implements a binary search tree."""
+
+
+class Node(object):
+    """Node object for use in a binary search tree."""
+
+    def __init__(self, value, right=None, left=None):
+        """Initialize a node for a binary search tree object."""
+        self.value = value
+        self.right = right
+        self.left = left
+
+    def _has_children(self):
+        """Return True or False if Node has children."""
+        if self.right or self.left:
+            return True
+        return False
+
+
+class BinarySearchTree(object):
+    """Binary Search Tree Object.
+
+    Methods:
+      - insert(self, val): will insert the value val into the BST. If val is already present, it will be ignored.
+      - search(self, val): will return the node containing that value, else None
+      - size(self): will return the integer size of the BST (equal to the total number of values stored in the tree). It will return 0 if the tree is empty.
+      - depth(self): will return an integer representing the total number of levels in the tree. If there is one value, the depth should be 1, if two values it will be 2, if three values it may be 2 or three, depending, etc.
+      - contains(self, val): will return True if val is in the BST, False if not.
+      - balance(self): will return an integer, positive or negative that represents how well balanced the tree is. Trees which are higher on the left than the right should return a positive value, trees which are higher on the right than the left should return a negative value. An ideally-balanced tree should return 0.
+
+    """
+
+    def __init__(self):
+        """Initialize a Binary Search Tree object."""
+        self.root = None
+        self.size = 0
+
+    def insert(self, val):
+        """Insert a new node with val into the BST."""
+        # if not self.search(val):
+        #     return
+        new_node = Node(val)
+        if self.root is None:
+            self.root = new_node
+            self.size = 1
+            return
+
+        curr_node = self.root
```

Comment:
minor nitpick, but you could just set curr to self.root in the beginning, then check if curr is none